src/main/scala/com/rbmhtechnology/eventuate/crdt/LWWRegister.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2015 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate.crdt
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import com.rbmhtechnology.eventuate.{Versioned, DurableEvent, VectorTime}
+
+import scala.concurrent.Future
+
+/**
+ * A LWWVersioned value.
+ *
+ * @param value The value.
+ * @param vectorTimestamp Timestamp of the event that caused this version.
+ * @param wallTimestamp Wall Clock
+ * @param creator Creator of the event that caused this version.
+ */
+case class LWWVersioned[A](value: A, vectorTimestamp: VectorTime, wallTimestamp: Long = 0L, creator: String = "")
+
+/**
+ * Replicated Last Writer Wins Register. Stores a single value and timestamp at which it was set. The value is updated
+ * only if the timestamp of the new value is greater than the timestamp of the current value.
+ * Note that this relies on synchronized system clocks. [[LWWRegister]] should only be used when the choice of
+ * value is not important for concurrent updates occurring within the clock skew.
+ *
+ * @param versionedValue Initially empty.
+ *
+ * @see [[http://hal.upmc.fr/docs/00/55/55/88/PDF/techreport.pdf A comprehensive study of Convergent and Commutative Replicated Data Types]]
+ */
+case class LWWRegister[A](versionedValue: Option[LWWVersioned[A]] = None) {
+
+  def value: Option[A] =
+    if(versionedValue.isDefined) Some(versionedValue.get.value) else None
+
+  def set(newValue: A, vectorTimestamp : VectorTime, wallTimestamp : Long, emitterId : String): LWWRegister[A] = {
+    if(versionedValue.isEmpty || (versionedValue.isDefined && (
+      // first compare the vector clocks
+      (vectorTimestamp > versionedValue.get.vectorTimestamp) ||
+        // if the vector clocks are concurrent, use the wall clock
+        ((vectorTimestamp conc versionedValue.get.vectorTimestamp) &&
+          (wallTimestamp > versionedValue.get.wallTimestamp)) ||
+        // If we still have a tie, choose the version with greatest emitter id
+        (wallTimestamp == versionedValue.get.wallTimestamp) &&
+          (emitterId > versionedValue.get.creator)))
+    )
+      copy(Some(LWWVersioned(newValue, vectorTimestamp, wallTimestamp, emitterId)))
+    else
+      copy(versionedValue)
+  }
+}
```

Comment:
I find that very difficult to read. What do you think about

```scala
case class LWWRegister[A](versionedValue: Option[LWWVersioned[A]] = None) {

  def value: Option[A] =
    versionedValue.map(_.value)

  def set(nvv: LWWVersioned[A]): LWWRegister[A] = {
    val result = versionedValue match {
      case None => 
        Some(nvv)
      case Some(vv) if nvv.vectorTimestamp > vv.vectorTimestamp => 
        Some(nvv)
      case Some(vv) if nvv.systemTimestamp > vv.systemTimestamp =>
        Some(nvv)
      case Some(vv) if nvv.systemTimestamp == vv.systemTimestamp && nvv.emitterId > vv.emitterId =>
        Some(nvv)
      case _ => 
        versionedValue
    }
    copy(result)
  }
}
```

together with a corresponding change in `LWWRegisterServiceOps`?